src/providers/mssql/qgsmssqldataitems.cpp

```diff
@@ -270,24 +251,29 @@ QVector<QgsDataItem *> QgsMssqlConnectionItem::createChildren()
       }
     }
 
-    // add missing schemas (i.e., empty schemas)
-    const QString uri = connInfo();
-    const QStringList allSchemas = QgsMssqlConnection::schemas( uri, nullptr );
-    for ( const QString &schema : allSchemas )
+    if ( !mUseGeometryColumns )
```

Comment:
I am not sure what is the reason of this change - why this code would be run only when `if ( !mUseGeometryColumns )` ?

Reply:
this piece of code search in all the schemas for those that are not already added (that is the empty one), but if we we have "only use geometry columns" checked we don't want all but only those that are in the geometry columns, no?

Reply:
I think this should stay independent from the "use geometry columns" flag... if there is a schema which is empty, we still want to show it (even if there are no tables/views in it), so the user is not confused that it does not exist